crates/nu-cli/src/commands/table.rs

```diff
@@ -291,7 +291,10 @@ async fn table(args: CommandArgs, registry: &CommandRegistry) -> Result<OutputSt
 
         let input: Vec<Value> = new_input.into();
 
-        if !input.is_empty() {
+        let table_is_in_default_empty_state = input.len() == 1 && input[0].is_none();
+        let should_draw_table = !input.is_empty() && !table_is_in_default_empty_state;
+
+        if should_draw_table {
```

Comment:
I wonder if we should move this up further. Before we even create this vec, we should know that the stream we're looking at is empty. If it is, we could probably avoid draining altogether (maybe?)

Reply:
What about the second case of the there being one row with nothing in it, would we be able to tell that from the perspective of an iterator?

Reply:
I was wondering if we could use this method on the stream:

https://github.com/nushell/nushell/blob/d30c40b40ebfbb411a503ad7c7bceae8029c6689/crates/nu-cli/src/stream/input.rs#L32-L34

Reply:
☝️ 👍  I added this specifically for the case where there's an empty value in the stream, but we  want to ignore it. This was important to know for running an external command, but we had to keep the empty value in the stream for everything else to know.

Another option is to rewrite everything to take an `Option<InputStream>`, but this could be a fairly extensive refactor (which I avoided with the `is_empty` trick).

Reply:
Right, @thegedge, I think Jonathan and I ended up finding a bug in InputStream related to this, which is what is allowing for this to happen.

There are several `from()` functions that set the value of the `empty` Boolean to `false`.  So even if the table is a single empty row, as soon as `from` is called, the `is_empty()` function will return `false`.  Secondly, there is no check for `is_empty()` in table.rs, there is simply a check to see if a final vector is empty of not, which never will be since the InputStream has the single empty row in it, so it always prints the table.

Reply:
> ☝️ 👍  I added this specifically for the case where there's an empty value in the stream, but we  want to ignore it. This was important to know for running an external command, but we had to keep the empty value in the stream for everything else to know.
> 
> 
> 
> Another option is to rewrite everything to take an `Option<InputStream>`, but this could be a fairly extensive refactor (which I avoided with the `is_empty` trick).

Also, that `Option<InputStream>` sounds neat, but certainly would be a massive change.

Reply:
Mind pointing me to a specific `from` instance that's problematic, and the context in which it's called?

Just an FYI, my intention when adding `empty` to `InputStream` was that it should be set to `false` _only_ when it's called via `InputStream::empty()`. I wanted it to be very explicit that the only case a stream is considered empty is through that function, because an input stream with a single `Nothing` is otherwise not empty.